### PR TITLE
Support extending elasticsearch AWS Request Signer

### PR DESCRIPTION
**Pull request by lucasdlemos**
- Support adding an additional module to `ElasticsearchConnectorFactory`
- Extract a new `ElasticsearchAwsModule` that provides a default implementation for the `AwsSignerCredentialsProvider` interface
- Change `ElasticsearchClient` to inject `Optional<AwsSignerCredentialsProvider>` rather than the AWS config, to allow extending AWS authentication with different credential providers

### CODE REVIEW

**Comment by kokosing** [comment]
@cla-bot check

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchConnectorFactory.java [hunk 1]
```diff
@@ -27,13 +29,24 @@
...
 import java.util.Map;
+import java.util.Optional;
 
 import static java.util.Objects.requireNonNull;
 
 public class ElasticsearchConnectorFactory
         implements ConnectorFactory
 {
-    ElasticsearchConnectorFactory() {}
+    private final Optional<Module> extension;
```
Comment (kokosing) [inline plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchConnectorFactory.java:39]: Module is a "collection" already, so you don't need to wrap it in list or optional. You can always use `Modules.EMPTY_MODULE` here. Then logic below is much easier.

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchConnectorFactory.java [hunk 2]
```diff
@@ -27,13 +29,24 @@
...
+
+    ElasticsearchConnectorFactory()
```
Comment (kokosing) [inline plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchConnectorFactory.java:41]: why not public?

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/aws/AwsRequestSigner.java [hunk 3]
```diff
@@ -11,7 +11,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package io.trino.plugin.elasticsearch.client;
+package io.trino.plugin.elasticsearch.aws;
```
Comment (kokosing) [inline plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/aws/AwsRequestSigner.java:14]: Please extract refactors (changes without logic changes) as separate commits.

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/aws/ElasticsearchAwsModule.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+    /**
+     * Provides a default Optional {@link HttpRequestInterceptor} to sign requests to an Elasticsearch cluster in AWS.
+     * If set, uses the configured aws access and security keys in {@link AwsSecurityConfig}, otherwise defaults to
+     * the {@link DefaultAWSCredentialsProviderChain}
+     *
+     * @param awsConfig aws security config
+     * @return AWS Request Signer
+     */
+    @ProvidesIntoOptional(DEFAULT)
```
Comment (kokosing) [inline plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/aws/ElasticsearchAwsModule.java:47]: Instead of using optional binder here, can we have `ElasticsearchAwsModule` to be a default extension in `ElasticsearchConnectorFactory`?

**Comment by findepi** [comment]
@lucasdlemos when this is useful?

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchConnectorFactory.java (hunk 1 above)
Comment (lucasdlemos) [reply plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchConnectorFactory.java]: Applied

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchConnectorFactory.java (hunk 2 above)
Comment (lucasdlemos) [reply plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchConnectorFactory.java]: This was not used externally, but with the new refactors I made it public.

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/aws/AwsRequestSigner.java (hunk 3 above)
Comment (lucasdlemos) [reply plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/aws/AwsRequestSigner.java:14]: Applied

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/aws/ElasticsearchAwsModule.java (hunk 4 above)
Comment (lucasdlemos) [reply plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/aws/ElasticsearchAwsModule.java:47]: I applied having the `ElasticsearchAwsModule` as a default extension, but we still want this binding to be `Optional` because it will only have a value if `elasticsearch.security` is set to `AWS`. 

This logic was already existing in `ElasticsearchClient`, so it knows when to sign requests with AWS credentials.

**Comment by lucasdlemos** [comment]
Thanks @kokosing, I applied your comments. Additionally I added a new commit to make it possible to extend the `BaseElasticsearchConnectorTest`.

**Comment by martint** [comment]
@lucasdlemos, what's the motivation for this change?

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchConnectorFactory.java [hunk 5]
```diff
@@ -28,12 +30,32 @@
...
+import static io.airlift.configuration.ConditionalModule.conditionalModule;
+import static io.trino.plugin.elasticsearch.ElasticsearchConfig.Security.AWS;
 import static java.util.Objects.requireNonNull;
+import static java.util.function.Predicate.isEqual;
 
 public class ElasticsearchConnectorFactory
         implements ConnectorFactory
 {
-    ElasticsearchConnectorFactory() {}
+    public static final Module DEFAULT_EXTENSION = conditionalModule(
```
Comment (kokosing) [inline plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchConnectorFactory.java:41]: wrap this `conditionalModule(...)` as named module class defined elsewhere.

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchConnectorFactory.java [hunk 6]
```diff
@@ -60,9 +82,8 @@ public Connector create(String catalogName, Map<String, String> config, Connecto
                 new JsonModule(),
                 new TypeDeserializerModule(context.getTypeManager()),
                 new ElasticsearchConnectorModule(),
-                binder -> {
```
Comment (kokosing) [inline plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchConnectorFactory.java:63]: undo formatting change.

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java [hunk 7]
```diff
@@ -53,11 +53,11 @@
 public abstract class BaseElasticsearchConnectorTest
         extends BaseConnectorTest
 {
-    private final String image;
-    private ElasticsearchServer elasticsearch;
+    protected final String image;
+    protected ElasticsearchServer elasticsearch;
```
Comment (kokosing) [inline plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java:57]: I think `BaseElasticsearchConnectorTest` should not hold infrastructure at all and so it should not define `createQueryRunner()` method at all. Reusing infrastructure code via inheritance may make code complex, it is hard then to understand what happens when.

Instead all implementation of `BaseElasticsearchConnectorTest` should define `createQueryRunner` on their own by using well defined building blocks to limit the amount of copy paste.

**Comment by kokosing** [comment]
> what's the motivation for this change?

We would like to customize the connector to allow it to use it as a library for other elastic search connector. The other connector is going to have different implementation of AWS authentication.

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchConnectorFactory.java (hunk 6 above)
Comment (lucasdlemos) [reply plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchConnectorFactory.java:63]: Applied

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java (hunk 7 above)
Comment (lucasdlemos) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java]: Applied. Moved building the server logic to the concrete classes.

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchConnectorModule.java [hunk 8]
```diff
@@ -43,16 +45,9 @@ protected void setup(Binder binder)
 
         configBinder(binder).bindConfig(ElasticsearchConfig.class);
 
-        newOptionalBinder(binder, AwsSecurityConfig.class);
+        newOptionalBinder(binder, Key.get(HttpRequestInterceptor.class, ForElasticsearchSigner.class));
```
Comment (martint) [inline plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchConnectorModule.java:48]: This is problematic. The fact that the connector uses Apache's HTTP client is an implementation detail. In the future, we plan to replace all of that with Airlift's HTTP client, so any "extensions" that depend on Apache HTTP client will break.

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchDefaultExtensionModule.java [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+{
+    @Override
+    protected void setup(Binder binder)
+    {
+        install(conditionalModule(
+                ElasticsearchConfig.class,
+                config -> config.getSecurity()
+                        .filter(isEqual(AWS))
+                        .isPresent(),
+                new ElasticsearchAwsModule()));
```
Comment (martint) [inline plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchDefaultExtensionModule.java:35]: Why is this a "default extension". The AWS authentication is a built-in feature of the elasticsearch connector.

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/ElasticsearchClient.java [hunk 10]
```diff
@@ -145,12 +141,12 @@
     @Inject
     public ElasticsearchClient(
             ElasticsearchConfig config,
-            Optional<AwsSecurityConfig> awsSecurityConfig,
+            @ForElasticsearchSigner Optional<HttpRequestInterceptor> awsRequestSigner,
```
Comment (martint) [inline plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/ElasticsearchClient.java:144]: ElasticsearchClient should not expose any details in its about the fact that it uses Apache HTTP client internally.

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchConnectorModule.java (hunk 8 above)
Comment (lucasdlemos) [reply plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchConnectorModule.java:48]: @martint would it be ok if we bind an `Optional<AwsCredentialsProvider>` and inject that in the `ElasticsearchClient`? This would also fix this comment: https://github.com/trinodb/trino/pull/8603#discussion_r674098907.

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchDefaultExtensionModule.java (hunk 9 above)
Comment (lucasdlemos) [reply plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchDefaultExtensionModule.java:35]: I initially let it as part of the `ElasticsearchConnectorModule` and changed to this to apply [this comment](https://github.com/trinodb/trino/pull/8603#discussion_r672907640). Should I move it back to `ElasticsearchConnectorModule`?

Comment (kokosing) [reply plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchDefaultExtensionModule.java:35]: That way we are able to replace the implementation of AWS authentication.

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java [hunk 11]
```diff
@@ -53,26 +46,9 @@
...
-            throws Exception
-    {
-        elasticsearch = new ElasticsearchServer(image, ImmutableMap.of());
-
-        HostAndPort address = elasticsearch.getAddress();
-        client = new RestHighLevelClient(RestClient.builder(new HttpHost(address.getHost(), address.getPort())));
-
-        return createElasticsearchQueryRunner(elasticsearch.getAddress(), TpchTable.getTables(), ImmutableMap.of(), ImmutableMap.of(), 3);
-    }
+    protected abstract void destroy();
```
Comment (kokosing) [inline plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java:51]: Why this is abstract method? This class does not call it.

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java [hunk 12]
```diff
@@ -14,21 +14,61 @@
...
         extends BaseElasticsearchConnectorTest
 {
+    private final ElasticsearchServer elasticsearch;
+    private final RestHighLevelClient client;
+
     public TestElasticsearch6ConnectorTest()
     {
-        super("docker.elastic.co/elasticsearch/elasticsearch-oss:6.0.0");
+        try {
+            elasticsearch = new ElasticsearchServer("docker.elastic.co/elasticsearch/elasticsearch-oss:6.0.0", ImmutableMap.of());
```
Comment (kokosing) [inline plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java:40]: Please do not setup resources in the constructor. Testng creates all the tests instances at once, and then run tests. That way we may end up having multiple elastic search containers to be up and running while some of them are not used.

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java [hunk 13]
```diff
@@ -14,21 +14,61 @@
...
+    {
+        return createElasticsearchQueryRunner(elasticsearch.getAddress(), TpchTable.getTables(), ImmutableMap.of(), ImmutableMap.of(), 3);
+    }
+
+    @AfterClass(alwaysRun = true)
+    @Override
+    public final void destroy()
+    {
+        try {
+            elasticsearch.stop();
```
Comment (kokosing) [inline plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java:66]: You can use `closeAfterClass` in `createQueryRunner` so destroy method might not be needed.

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java (hunk 11 above)
Comment (lucasdlemos) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java]: Removed it and got rid of the destroy method using `closeAfterClass` as you suggested.

**Comment by lucasdlemos** [comment]
@kokosing I applied you test comments in a new fixup commit: https://github.com/trinodb/trino/pull/8603/commits/5c4c95d1e774cb9a2ed94d4f1d8312ee708b0749.

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java (hunk 12 above)
Comment (lucasdlemos) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java]: Applied

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java (hunk 13 above)
Comment (lucasdlemos) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java]: Applied

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchConnectorModule.java (hunk 8 above)
Comment (lucasdlemos) [reply plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchConnectorModule.java]: We would also need the aws region. I created a new interface `AwsSignerCredentialsProvider` and provided a default implementation to it, so others can extend it with custom `AwsCredentialsProvider`. Please check this fixup commit with the changes: https://github.com/trinodb/trino/pull/8603/commits/85b7e752fb9e0047c6fe941f0b3c808df7bf1252

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchDefaultExtensionModule.java (hunk 9 above)
Comment (lucasdlemos) [reply plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchDefaultExtensionModule.java:35]: @kokosing since we're using an Optional binder and providing a default implementation to it in `ElasticsearchAwsModule` it'd still be possible to install that module in `ElasticsearchConnectorModule` and allow others to override the optional by calling `setBinding()`. 

I kept this separate "default extension" module for now but please let me know if I should remove it or not.

cc @martint

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/ElasticsearchClient.java (hunk 10 above)
Comment (lucasdlemos) [reply plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/ElasticsearchClient.java]: Changed it to inject `Optional<AwsSignerCredentialsProvider>` instead.

**Comment by kokosing** [comment]
I am sorry, but I am getting lost in these fixup commits. Can you please squash them?

**Comment by lucasdlemos** [comment]
> I am sorry, but I am getting lost in these fixup commits. Can you please squash them?

Sure @kokosing. I just squashed them.

**Comment by hashhar** [comment]
@martint the reason the CLA-bot isn't triggerring is because the Git identity of the committer has no email associated.

@lucasdlemos Can you please correct it and add an email to your commits - it shows up as `Author: Lucas Lemos <>` for me.

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/ElasticsearchClient.java [hunk 14]
```diff
@@ -206,7 +203,7 @@ private void refreshNodes()
 
     private static BackpressureRestHighLevelClient createClient(
             ElasticsearchConfig config,
-            Optional<AwsSecurityConfig> awsSecurityConfig,
+            Optional<AwsSignerCredentialsProvider> awsSignerConfig,
```
Comment (kokosing) [inline plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/ElasticsearchClient.java:206]: update variable name

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/ElasticsearchClient.java [hunk 15]
```diff
@@ -246,26 +243,16 @@ private static BackpressureRestHighLevelClient createClient(
                 clientBuilder.setDefaultCredentialsProvider(credentials);
             });
 
-            awsSecurityConfig.ifPresent(securityConfig -> clientBuilder.addInterceptorLast(new AwsRequestSigner(
-                    securityConfig.getRegion(),
-                    getAwsCredentialsProvider(securityConfig))));
+            awsSignerConfig
+                    .map(awsConfig -> new AwsRequestSigner(awsConfig.getAwsRegion(), awsConfig.getCredentialsProvider()))
+                    .ifPresent(clientBuilder::addInterceptorLast);
```
Comment (kokosing) [inline plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/ElasticsearchClient.java:248]: Instead of having optional binder for `AwsSignerCredentialsProvider`. Would it be possible to have a multi set binder for `HttpRequestInterceptor` so we could just call `addInterceptorLast()` for each of them in order they are binded in guice? Then you would provide such in `ElasticsearchAwsModule`.

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/ElasticsearchServer.java [hunk 16]
```diff
@@ -68,8 +73,21 @@ public void stop()
         deleteRecursively(configurationPath, ALLOW_INSECURE);
     }
 
+    @Override
+    public void close()
+            throws IOException
+    {
+        stop();
```
Comment (kokosing) [inline plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/ElasticsearchServer.java:80]: do we still need public `stop()` method? (separate commit to make `ElasticsearchServer` closeable?)

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java [hunk 17]
```diff
@@ -162,6 +162,7 @@ public void testSortItemsReflectedInExplain()
                 "TopNPartial\\[5 by \\(nationkey DESC");
     }
 
+    @Test
```
Comment (kokosing) [inline plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java:165]: Was it causing any issues? Maybe we should drop other `@Test` if they are not needed instead?

**Review by kokosing — APPROVED** [review]

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/ElasticsearchClient.java (hunk 15 above)
Comment (lucasdlemos) [reply plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/ElasticsearchClient.java:248]: @kokosing This is close to what I originally had, although I applied [this comment](https://github.com/trinodb/trino/pull/8603#discussion_r674098907) from Martin because we want to keep the fact that `ElasticsearchClient` uses the Apache Http Client as an implementation detail and not expose it externally.

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/ElasticsearchServer.java (hunk 16 above)
Comment (lucasdlemos) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/ElasticsearchServer.java]: Got rid of `stop()` and extracted the change to a separate commit.

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java (hunk 17 above)
Comment (lucasdlemos) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java:165]: The `@Test` annotation is still needed in the super class for methods that don't override this method. 

I added this annotation here because otherwise it causes issues if another class extending this test suite uses JUnit5 and has to parse testng test methods.

**Comment by lucasdlemos** [comment]
> @martint the reason the CLA-bot isn't triggerring is because the Git identity of the committer has no email associated.
> 
> @lucasdlemos Can you please correct it and add an email to your commits - it shows up as `Author: Lucas Lemos <>` for me.

Oh, I had it misconfigured. Sorry about that @hashhar. I fixed it now and the CLA bot now works.

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java [hunk 18]
```diff
@@ -53,26 +46,7 @@
...
-    protected QueryRunner createQueryRunner()
-            throws Exception
-    {
-        elasticsearch = new ElasticsearchServer(image, ImmutableMap.of());
-
-        HostAndPort address = elasticsearch.getAddress();
-        client = new RestHighLevelClient(RestClient.builder(new HttpHost(address.getHost(), address.getPort())));
-
-        return createElasticsearchQueryRunner(elasticsearch.getAddress(), TpchTable.getTables(), ImmutableMap.of(), ImmutableMap.of(), 3);
-    }
```
Comment (sumannewton) [inline plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java:75]: Why is this moved to subclasses?

Comment (hashhar) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java:75]: Abstract test classes shouldn't dictate how the query runner is set up. It's better to be explicit and let each subclass define their own runners. Makes it clear if the test depends on some specific configuration.

That's how we've been doing in other connectors - this one was an outlier.

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestPasswordAuthentication.java [hunk 19]
```diff
@@ -85,7 +85,7 @@ public final void destroy()
     {
         closeAll(
                 () -> assertions.close(),
-                () -> elasticsearch.stop(),
+                () -> elasticsearch.close(),
```
Comment (kokosing) [inline plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestPasswordAuthentication.java:88]: do you need a lambda here?

**Review by kokosing — APPROVED** [review]

File: plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestPasswordAuthentication.java (hunk 19 above)
Comment (lucasdlemos) [reply plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestPasswordAuthentication.java]: Indeed it was not needed anymore.

**Comment by martint** [comment]
> We would like to customize the connector to allow it to use it as a library for other elastic search connector. The other connector is going to have different implementation of AWS authentication.

I'm not convinced this is a good idea. The Elasticsearch connector was never designed to be a library, and doing that places too many constraints on the implementation -- now we have to think about what's part of the library API, what we can change without breaking implementations that depend on it, etc.

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/aws/AwsSignerCredentialsProvider.java [hunk 20]
```diff
@@ -0,0 +1,23 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.elasticsearch.aws;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+
+public interface AwsSignerCredentialsProvider
```
Comment (martint) [inline plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/aws/AwsSignerCredentialsProvider.java:18]: Why is this an interface? How many kinds of AWS credentials providers can exist?

**Comment by lucasdlemos** [comment]
Closing this PR as it was decided this extension won't be needed for now.
